### PR TITLE
DOC: Enhance the README.md syntax.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Make it consistent with the ITK Markdown file style (bullets, indentation,
etc.)

Add links to external tools and packages (CMake, Python, etc.)

Fix the code syntax highlighting error in the `Building with Docker`
section (legacy from wiki syntax)

Remove the sentence
`TODO: These options are missing from msg\_gerrit() in
./git/hooks/commit-msg file.`
An issue will be openend in the InsightSoftwareConsortium project

Pointed to specific GitHub files and folders for ITK SW Guide elements.

Use syntax highlighting extensively to improve readability.

Change-Id: I31178a1286a2e8c1c11b693acaba409eac0e88d6

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Is this right
`PNG files are generated by running ITK examples and converted to EPS using MagicTools; the resulting EPS files are saved in Art/Generated directory of the binary output directory.`
?

Can anybody provide a link to MagicTools?

Or do we mean ImageMagick?

**Comment by jhlegarreta** [comment]
BTW,
`LaTeX/00-SoftwareGuide.tex`
Does not exist.
May be referring to
`SoftwareGuide/Latex/00-Preamble-Common.tex`
?
But these
`Latex/Insight.sty`
`Latex/InsightSoftwareGuide.cls`
also have soem package requirements

**Comment by thewtex** [comment]
> Or do we mean ImageMagick?

Yes.

File: README.md [hunk 1]
```diff
@@ -1,188 +1,192 @@
...
 
 Build Overview
 --------------
 
-The generation of the ITK Software Guide is orchestrated as a CMake superubild
-process. CMakeLists.txt files are placed in the directories involved on the
-build process. As any other CMake-managed process, the results of the build
-process are put in a binary tree corresponding to the source tree.
+The generation of the ITK Software Guide is orchestrated as a [CMake]
+`Superubild` process. `CMakeLists.txt` files are placed in the directories
```
Comment (thewtex) [inline README.md]: Superbuild

**Comment by thewtex** [comment]
> SoftwareGuide/Latex/00-Preamble-Common.tex

Yes, content was migrated here after splitting into two books.

**Comment by hjmjohnson** [comment]
@jhlegarreta Conflicts exist.  Please rebase.

**Comment by jhlegarreta** [comment]
@hjmjohnson Thanks for the keeping an eye on this. Commit ee89a28 rebased the branch.

**Pull request merged**